api/v1alpha1/driver_types.go

```diff
@@ -0,0 +1,271 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type LogSpec struct {
+	// Log level for driver pods,
+	// Supported values from 0 to 5. 0 for general useful logs (the default), 5 for trace level verbosity.
+	// Default to 0
+	LogLevel   int `json:"logLevel,omitempty"`
+	MaxFiles   int `json:"maxFiles,omitempty"`
+	MaxLogSize int `json:"maxLogSize,omitempty"`
```

Comment:
in which format it is Gib or Gb or Mib? we might need to revisit and use resource.Quality for this one.

Reply:
Why `resource.Quality`? isn't that overengineering? Can't we just decide on a specific unit and stick to it. We can even reflect it in the fields names

```suggestion
	MaxLogSizeInGiB int `json:"maxLogSize,omitempty"`
```

Reply:
makes sense but this is forcing the user to push to have min 1 Gib (which is not good) as a user I don't want to wait for that why can't I have in Mib, I don't want to have a single file with 1 Gib but i do want to have max 500Mb of file and rotate is periodically,

Reply:
The GiB was only an example, we can use MiB instead. But if you strongly see the need for flexibility I will change to `resource.Quantity`

Reply:
Yes lets go for resource.Quantity as it provides the flexibility for us

Reply:
done